packages/net/ssl/ssl.pony

```diff
@@ -147,7 +151,22 @@ class SSL
     if ready then
       _read_buf = recover Array[U8] end
     else
-      None
+      // try and read again any pending data that SSL hasn't decoded yet
+      if @BIO_ctrl_pending[USize](_input) > 0 then
+        read(expect)
+      else
+        ifdef "openssl_1.1.x" then
```

Comment:
this is openssl 1.1 specific?  what did they change that requires this?

Reply:
Yes, `SSL_has_pending` is new in `1.1`. This mailing list post has a good description of what it is for: https://mta.openssl.org/pipermail/openssl-users/2017-January/005110.html

Reply:
Could you add a link to that in the code? I think it would be quite handy in the future.